core/trino-main/src/main/java/io/trino/sql/planner/PlanFragmenter.java

```diff
@@ -323,9 +323,6 @@ public PlanNode visitRefreshMaterializedView(RefreshMaterializedViewNode node, R
         @Override
         public PlanNode visitTableWriter(TableWriterNode node, RewriteContext<FragmentProperties> context)
         {
-            if (node.getPartitioningScheme().isPresent()) {
-                context.get().setDistribution(node.getPartitioningScheme().get().getPartitioning().getHandle(), metadata, session);
-            }
             return context.defaultRewrite(node, context.get());
```

Comment:
why not:
```
context.defaultRewrite(node, context.get()); // RemoteExchange will set scaled distribution
if (node.getPartitioningScheme().isPresent()) {
  context.get().setDistribution(...)
}
```

in `setDistribution` you can check that the distribution you try to set is same as pre-set one and accepts scaling of writers.